packages/cli/src/utils.ts

```diff
@@ -91,18 +92,61 @@ export async function processFiles(
   fn: AsyncWorker<string>,
   fileCountFn = (count: number) => {}
 ): Promise<void> {
-  const q = queue(fn, 5);
-  q.pause();
   const files = await promisify(glob)(pattern);
   if (fileCountFn) {
     fileCountFn(files.length);
   }
   if (files.length === 0) return;
+
+  const q = queue(fn, 2);
   files.forEach((file) => q.push(file));
-  q.resume();
   await q.drain();
 }
 
+/**
+ * Finds all occurrances of `fileName` within a base directory. Each match must be a file, not a directory.
+ * This is optimized compared to `processFiles` because it does not use `glob`, which is pretty slow for this use case.
+ * It also begins processing right away, rather than waiting for all files to be enumerated.
+ */
+export async function processNamedFiles(
+  baseDir: string,
+  fileName: string,
+  fn: AsyncWorker<string>
+): Promise<void> {
+  const q = queue(fn, 2);
+
+  const stats = async (fileName: string): Promise<Stats | undefined> => {
+    try {
+      return await stat(fileName);
+    } catch {
+      // Ignore
+    }
+  };
+
+  let matchCount = 0;
+  const processDir = async (dir: string) => {
+    // If the directory contains the target fileName, add it to the process queue and return.
+    const targetFileName = join(dir, fileName);
+    const target = await stats(targetFileName);
+    if (target && target.isFile()) {
+      matchCount += 1;
+      q.push(targetFileName);
+      return;
+    }
+
+    // Otherwise, recurse on all subdirectories.
+    for (const childName of await readdir(dir)) {
+      const child = await stats(join(dir, childName));
+      if (child && child.isDirectory()) {
+        await processDir(join(dir, childName));
+      }
+    }
+  };
+  await processDir(baseDir);
+
+  if (matchCount) await q.drain();
```

Comment:
Perhaps it'd be better to check `q.idle()` instead — I think `q.drain()` might hang if the queue has emptied in the meantime.

Reply:
Yeah this has always been an issue with these queues. I'll change it `drain unless q.idle()`

Reply:
I've done this project-wide, let me know what you think.